extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -96,20 +123,33 @@ export class ScrollManager {
 
   /**
    * @param {!../../../src/service/viewport/viewport-interface.ViewportChangedEventDef} e
+   * @return {!Promise}
    * @private
    */
   onScroll_(e) {
-    /** {./scroll-manager.ScrollEventDef} */
-    const scrollEvent = {
-      top: e.top,
-      left: e.left,
-      width: e.width,
-      height: e.height,
-      scrollWidth: this.viewport_.getScrollWidth(),
-      scrollHeight: this.viewport_.getScrollHeight(),
-    };
-    // Fire all of our children scroll observables
-    this.scrollObservable_.fire(scrollEvent);
+    return this.measureRootElement_(true /** initial */).then(() => {
+      devAssert(this.initialRootLayoutRect_);
+      const {
```

Comment:
- Shouldn't `/* initial */ true` return `this.initialRootLayoutRect_`?
- Check initial at first to prevent double measure. (Unless you don't need the first measure?)

```js

const hadInitial = !this.initialRootLayoutRect_;

this.getOrMeasureInitialRootRect_().then(initialRect => {
  const initialScrollLength = {
    scrollWidth: initialRect.width,
    scrollHeight: initialRect.height,
  };
  const currentRect = hadInitial ? this.measureRootElement_() : Promise.resolve(initialRect);
  currentRect.then(layoutRect => {
    const scrollEvent = {
      initialScrollLength,
      // ...
    };
    // ...
  });
});
```